### PR TITLE
Documents noargs HELLO variant

### PR DESCRIPTION
**Pull request by itamarhaber**
Ref: https://github.com/redis/redis/pull/7377

image

### CODE REVIEW

File: commands/hello.md [hunk 1]
```diff
@@ -21,20 +45,25 @@ RESP2). Clients that want to handshake the RESP3 mode need to call the
...
-ID, the replication role and so forth. Because of that, and given that
-the `HELLO` command also works with "2" as argument, both in order to
-downgrade the protocol back to version 2, or just to get the reply from
-the server without switching the protocol, client library authors may
-consider using this command instead of the canonical `PING` when setting
-up the connection.
+In both cases, `HELLO`'s reply states several useful facts about the server,
+such as: versions, modules loaded, client ID, replication role and so forth.
+Because of that, and given that the `HELLO` command also works with "2" as an
+argument - both to downgrade the protocol back to version 2, or just to get the
```
Comment (oranagra) [inline commands/hello.md:51]: This sentence still seems to be mixing between value of 2 as a downgrade, and value of 2 as a call to HELLO "without switching" version

**Comment by oranagra** [comment]
btw https://github.com/redis/redis-doc/blob/master/README.md says:
```
Wrap lines to 80 characters.
Start every sentence on a new line.
```

i have no idea why

**Comment by itamarhaber** [comment]
> i have no idea why

Me neither, but I'm not pedantic about it (as prior art isn't as well).

File: commands/hello.md [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
+replies: for instance, `HGETALL` will return a *map type*, so a client library
+implementation no longer requires to know in advance to translate the array into
+a hash before returning it to the caller. For a full coverage of RESP3, please
 [check this repository](https://github.com/antirez/resp3).
 
 In Redis 6 connections start in RESP2 mode, so clients implementing RESP2 do
-not need to updated or changed (nor there are short term plans to drop support for
-RESP2, but future version may default to RESP3).
+not need to updated or changed. There are short term plans to drop support for
+RESP2, although future version may default to RESP3.
```
Comment (oranagra) [inline commands/hello.md:14]: did you notice you negated the sentence?
```suggestion
not need to updated or changed. There are no short term plans to drop support for
RESP2, although future version may default to RESP3.
```

File: commands/hello.md [hunk 3]
```diff
@@ -45,17 +46,13 @@ command with the value "3" as first argument, like so:
...
-In both cases, `HELLO`'s reply states several useful facts about the server,
-such as: versions, modules loaded, client ID, replication role and so forth.
-Because of that, and given that the `HELLO` command also works with "2" as an
-argument - both to downgrade the protocol back to version 2, or just to get the
-reply from the server without switching the protocol - client library authors
-may consider using this command instead of the canonical `PING` when setting up
-the connection.
+Because `HELLO` replies with useful information, and given that `protover` is
+optional and can be set to "2" (both to downgrade the protocol back to version
+2 or ), client library authors may consider using this command instead of the
```
Comment (oranagra) [inline commands/hello.md:51]: `to version 2 or )` ??

File: commands/hello.md (hunk 2 above)
Comment (itamarhaber) [reply commands/hello.md:14]: No, I haven't (noticed).

**Review by oranagra — APPROVED** [review]

**Pull request merged**